src/SimpleGraphs/generators/randgraphs.jl

```diff
@@ -989,6 +989,144 @@ function uniform_tree(n::Integer; rng::Union{Nothing,AbstractRNG}=nothing)
     return prufer_decode(random_code)
 end
 
+"""
+    havel_hakimi_graph_generator(degree_sequence::AbstractVector{<:Integer})
+
+Returns a simple graph with a given finite degree sequence of non-negative integers generated via the Havel-Hakimi algorithm which works as follows:
+1. successively connect the node of highest degree to other nodes of highest degree;
+2. sort the remaining nodes by degree in decreasing order;
+3. repeat the procedure.
+
+## References
+1. [Hakimi (1962)](https://doi.org/10.1137/0110037);
+2. [Wikipedia](https://en.wikipedia.org/wiki/Havel%E2%80%93Hakimi_algorithm).
+"""
+function havel_hakimi_graph_generator(degree_sequence::AbstractVector{<:Integer})
+    # Check whether the degree sequence has only non-negative values
+    all(degree_sequence .>= 0) ||
+        throw(ArgumentError("The degree sequence must contain non-negative integers only."))
```

Comment:
Would it make sense, to just call `isgraphical` here? Or do you think that the overhead is too big then?

Reply:
Maybe, in order to maximize performance and code simplicity, we should completely separate the functionalities of  `isgraphical` and `havel_hakimi_graph`.  So if the user knows that the sequence is graphical and just wants the graph, she/he may use `havel_hakimi_graph`, while if just/also a check for graphicality is needed, then a call to `isgraphical` must be made.

Then I'd suggest to either:

1. Remove all checks from `havel_hakimi_graph`;
2. Add a bool kwarg `check_graphicality` to `havel_hakimi_graph`, that performs the `isgraphical` check before executing the algorithm (in which case we may skip the subsequent check in the loop).

Similar reasoning would apply to `isdigraphical` and `kleitman_wang_graph`.